posthog/session_recordings/session_recording_api.py

```diff
@@ -539,10 +539,37 @@ def list_recordings(
     with timer("load_persons"):
         # Get the related persons for all the recordings
         distinct_ids = sorted([x.distinct_id for x in recordings])
+        # persons can have an effectively unbounded number of distinct ids
+        # we are only going to return ten
+        # so, we don't need to load them all
+        # in the wild we've seen customers with a high number of distinct ids
+        # per person have very slow responses _only_ from loading distinct ids per person
+
+        # So first, we create a subquery to fetch the limited distinct IDs for a given person.
+        limited_distinct_ids_subquery = (
+            PersonDistinctId.objects.filter(person=OuterRef("person")).order_by("id").values("distinct_id")[:10]
+        )
+
+        # Then we create a query that only loads person distinct IDs for the limited distinct IDs.
+        # that might seem silly, and it sort of is
+        # Django doesn't let us slice the query in a prefetch
+        # but, we can slice a query in a query in a prefetch :shrug:
+        limited_distinct_ids = PersonDistinctId.objects.filter(
+            distinct_id__in=Subquery(limited_distinct_ids_subquery.values("distinct_id")), team=team
+        )
+
+        # Now, prefetch these limited distinct IDs when fetching the PersonDistinctId objects.
+        # We want to prefetch because we're going to be loading a page of 20 people
+        # and don't want to make 20 queries against the DB
+        # this _should_ make three queries
+        # before we had the nested queries we would make two queries
+        # but they could be very slow... these should be faster
         person_distinct_ids = (
             PersonDistinctId.objects.filter(distinct_id__in=distinct_ids, team=team)
             .select_related("person")
-            .prefetch_related(Prefetch("person__persondistinctid_set", to_attr="distinct_ids_cache"))
+            .prefetch_related(
+                Prefetch("person__persondistinctid_set", queryset=limited_distinct_ids, to_attr="distinct_ids_cache")
+            )
```

Comment:
I couldn't fully understand what the code is doing tbh.

In my mind we are saying:
1. Recording has distinct_id `A`
2. Go via the joining table to find Person with distinct_id `A`
3. Return that Person.

What I don't get it is what all this code does (the existing code and the new code) 😅 
We seem to be getting all PersonDistincIds (the joining table) with the Person _and_ all their other distinct_ids (wat?), and then looping over to to find the person? 

Could we not just remove the prefetch entirely?